core/keyserver/keyserver.go

```diff
@@ -358,7 +360,10 @@ func (s *Server) GetDomainInfo(ctx context.Context, in *tpb.GetDomainInfoRequest
 		return nil, grpc.Errorf(codes.InvalidArgument, "Please specify a domain_id")
 	}
 	domain, err := s.admin.Read(ctx, in.DomainId, false)
-	if err != nil {
+	if err == sql.ErrNoRows {
+		glog.Errorf("adminstorage.Read(%v): %v", in.DomainId, err)
+		return nil, status.Errorf(codes.NotFound, "Domain %v not found", in.DomainId)
+	} else if err != nil {
 		glog.Errorf("adminstorage.Read(%v): %v", in.DomainId, err)
 		return nil, grpc.Errorf(codes.Internal, "Cannot fetch domain info")
```

Comment:
don't you want to say anything about in.DomainId in this error message?

Reply:
done